src/mcts/search.cc

```diff
@@ -1291,9 +1302,20 @@ void SearchWorker::ExtendNode(Node* node) {
       return;
     }
 
-    if (history_.Last().GetRepetitions() >= 2) {
+    const auto repetitions = history_.Last().GetRepetitions();
+    const auto twofolddraws = params_.GetTwoFoldDraws();
```

Comment:
This is only accessed once in the `if`, better move it there.

Reply:
Agreed. Is it also possible to calculate `cyclelength` within the if statement and use it later? Right now, it is accessed twice for actual twofold draws.

Reply:
I hope the compiler will optimize it, but probably needs a `const` qualifier for that.

Reply:
This seems to be the case according to @Mardak. Marking this as resolved then.